watchface

```diff
@@ -0,0 +1,406 @@
+        echo "OK!"
+        if [ "${action}" = "clone" ] ; then
+            if newface=$("${MENUPROGRAM}" \
+                --title "Cloning ${watchface}" --clear \
+                --inputbox "Enter name of new watchface:" \
+                8 50 "newface" \
+                3>&1 1>&2 2>&3)
+            then
+                "${action}"face "${watchface}" "${newface}"
+            else
+                echo "No name entered"
+            fi
+        elif [ "${action}" = "deploy" ] ; then
+            local first=true
+            for wf in ${watchface}; do
+                "${action}"face "${wf}" "${first}"
+                first=false
+            done
+            if [ "${REBOOT}" = "true" ] ; then
+                rebootWatch
+            fi
+        else
+            "${action}"face "${wf}"
+        fi
+    else
+        echo "Canceled"
+    fi
+}
+
+# These are the defaults for SSH access
+WATCHPORT=22
+WATCHADDR=192.168.2.15
+# These are the defaults for local QEMU target
+QEMUPORT=2222
+QEMUADDR=localhost
+# Assume no ADB unless told otherwise
+ADB=false
+# Assume no GUI unless told otherwise
+GUI=false
+# we're not done until we say we are
+DONE=false
+# No wallpaper unless asked
+WALLPAPER=""
+# Assume Dialog unless unavailable
+if hash dialog 2>/dev/null; then
+    MENUPROGRAM=dialog
+else
+    MENUPROGRAM=whiptail
+fi
+
+
+while [[ $# -gt 0 ]] ; do
+    case $1 in
+        -a|--adb)
+            ADB=true
+            shift
+            ;;
+        -b|--boot)
+            REBOOT=true
+            shift
+            ;;
+        -g|--gui)
+            GUI=true
+            shift
+            ;;
+        -q|--qemu)
+            WATCHPORT=${QEMUPORT}
+            WATCHADDR=${QEMUADDR}
+            shift
+            ;;
+        -p|--port)
+            WATCHPORT="$2"
+            shift
+            shift
+            ;;
+        -r|--remote)
+            WATCHADDR="$2"
+            shift
+            shift
+            ;;
+        -h|--help)
+            showHelp
+            exit 1
+            ;;
+        -w|--wall)
+            WALLPAPER="$2"
+            shift
+            shift
+            ;;
+        clone)
+            cloneface "$2" "$3"
+            shift
+            shift
+            shift
+            DONE=true
+            ;;
+        deploy)
+            deployface "$2" true
+            shift
+            shift
+            DONE=true
+            ;;
+        test)
+            testface "$2"
+            shift
+            shift
+            DONE=true
+            ;;
+        version)
+            showVersion
+            shift
+            DONE=true
+            ;;
+        *)
+            echo "Ignoring unknown option $1"
+            shift
+            ;;
+    esac
+done
+
+mapfile -t WATCHFACES < <(find . -maxdepth 3 -type d -path "*/usr/share" -printf "%h\n" | sort | awk -F '/' '{print $2}' )
+if [ "${DONE}" != "true" ] ; then
+    if [ "${GUI}" = "true" ] ; then
+        doGUI
+    else
+        textmenu
```

Comment:
To keep things consistent, let's call these functions guiMenu and textMenu

Reply:
Agreed.  Fixed!